### DISCUSSION TITLE
SO Cases - Reporting

### DISCUSSION THREAD

**Discussion by Hammer-Bear** [Security-Onion-Solutions/securityonion#7225] [Unsupported Versions] [answered]
Hi,

So been using the new cases function in SO and I'm finding it much more fluid than the hive but one thing it seems to be lacking is the reporting side of things, you can filter by tags but that only gets me so far. Is there any plans to further develop the built-in cases function? Would be nice to do something in Kibana for our clients who like the pretty graphs and stats etc,

**Comment by dougburks** [comment, marked as answer]
> So been using the new cases function in SO and I'm finding it much more fluid than the hive

Thanks, glad you like it!

> Is there any plans to further develop the built-in cases function?

Yes, we do plan to add features to Cases as time goes on.

> Would be nice to do something in Kibana for our clients who like the pretty graphs and stats etc,

You should be able to visualize Cases data in your own custom Kibana dashboards today by searching the `so-case` index:
```
_index:"*:so-case"
```

**Reply by Hammer-Bear** [reply]
Awesome thanks, i had a look through kibana and I can see the so-case index and I can create a new index pattern but if I use the so-* index pattern it won't bring up the relevant fields and all appear empty. Also, some of the fields don't appear to be useable in visualisations according to the index fields

**Reply by dougburks** [reply]
Seems to be working for me:
![image](https://user-images.githubusercontent.com/1659467/155414960-818cb034-931e-479b-b5b2-5b48ff9664b6.png)
![image](https://user-images.githubusercontent.com/1659467/155415528-abb64687-f6ec-4f7d-9d57-9b3ede171a6d.png)
![image](https://user-images.githubusercontent.com/1659467/155415742-86cd2eca-6cd9-4da4-a3e7-d5f7ed2c9117.png)

**Reply by Hammer-Bear** [reply]
So far the ones I'm having issues with are the title and desc, looking at the index patterns it's saying these 2 fields aren't aggregatable.

![image](https://user-images.githubusercontent.com/36236670/155503458-91800022-a5d3-431d-9b10-6c3e2f97f41a.png)

**Reply by dougburks** [reply]
`so_case.description` and `so-case.title` are intentionally `text` (not aggregatable) to reduce load on Elasticsearch. One option might be to build a dashboard that represents the aggregatable fields as data tables but then include a saved search panel at the bottom that displays the main fields from the case including the non-aggregatable ones.

**Comment by pboosten** [comment]
My wishlist for cases would be that I can create a report of a case to send to our customers, in a nice template (of our own choosing).

We've changed SO to be multi-tenant, so we're going to create cases for different customers. An API to extract information from cases would suffice, actually.

We don't do remediation, just reporting.

Peter
